### PR TITLE
Update to Kubernetes v1.31.4

### PR DESCRIPTION
**Pull request by mikkeloscar**
This updates the AMI to Kubernetes v1.31.4: https://github.com/kubernetes/kubernetes/releases/tag/v1.31.4

Additionally it makes the following changes:

* Switch to upstream version of kube-controller-manager. This is possible because the fix for https://github.com/kubernetes/kubernetes/issues/127792 is part of Kubernetes v1.31.4
* Add the CNI config to the standard path `/etc/cni/net.d/` this is following a change in the AMI to expect the config at that path. This is done to be compatible with AWS CNI plugin which is expecting the standard path. We need later a follow up to remove the legacy path from the userdata. Added a TODO comment to cover this.

### CODE REVIEW

File: cluster/node-pools/master-default/userdata.yaml [hunk 1]
```diff
@@ -928,6 +928,7 @@ write_files:
       {{- end}}
 {{- end}}
 
+  # TODO: Remove this once all nodes are running an AMI compatible with /etc/cni/net.d/10-flannel.conflist
```
Comment (demonCoder95) [inline cluster/node-pools/master-default/userdata.yaml:931]: Do we cover this in some internal issue? Would be better recorded there. I'm guessing we have an issue for the VPC CNI work?

Comment (mikkeloscar) [reply cluster/node-pools/master-default/userdata.yaml:931]: I created an internal issue: 3756

Comment (demonCoder95) [reply cluster/node-pools/master-default/userdata.yaml:931]: Thanks 🙏

**Comment by demonCoder95** [comment]
I think the image wasn't promoted yet to the registry when the pipeline ran, so it failed

```
time="2024-12-11T14:13:22Z" level=fatal msg="Fail to provision: unable to read configuration defaults: template: main/cluster/config-defaults.yaml:743:38: executing \"main/cluster/config-defaults.yaml\" at <amiID \"zalando-ubuntu-jammy-22.04-kubernetes-production-v1.31.4-amd64-master-359\" \"861068367966\">: error calling amiID: no image found with name: zalando-ubuntu-jammy-22.04-kubernetes-production-v1.31.4-amd64-master-359 and owner: 861068367966"
```

The name does appear correct comparing to the AMI pipeline, so I just retriggered.

**Comment by demonCoder95** [comment]
Seems like that was it, e2e passes now.

**Comment by demonCoder95** [comment]
👍

**Comment by linki** [comment]
👍

**Pull request merged**